securedrop_client/db.py

```diff
@@ -77,30 +77,51 @@ class Submission(Base):
     download_url = Column(String(255), nullable=False)
 
     # This is whether the submission has been downloaded in the local database.
-    is_downloaded = Column(Boolean(name='is_downloaded'),
-                           default=False)
+    is_downloaded = Column(Boolean(name='is_downloaded'), nullable=False, server_default='false')
 
     # This reflects read status stored on the server.
-    is_read = Column(Boolean(name='is_read'),
-                     default=False)
+    is_read = Column(Boolean(name='is_read'), nullable=False, server_default='false')
+
+    content = Column(
```

Comment:
we can add this `content` column now, but if so let's not populate it until #217 (small PRs will get merged faster so let's keep these tickets separate)

in terms of the check constraint, I'm not sure this is what we want - here are the states I was expecting for a message/reply:

(initial state) 1. message `is_downloaded = false`, content is null

2a. download success, message `is_downloaded = true` encrypted file stored on disk in `filename` (next state) - content is still null (i.e. this violates the check constraint here)

2b. download failed, go to 1

3a. message `is_downloaded = true`, decryption is successful, content is not null

3b. decryption failed, go to 2

Reply:
Oh derp, that is definitely right. I would have noticed that once I started working on it. :sweat_smile: So that constraint should just be `CHECK is_downloaded = FALSE THEN content IS NULL ELSE TRUE END` to keep things in a predictable state.

Reply:
yep that sounds good!